lib/config/fileconf.go

```diff
@@ -991,6 +991,15 @@ type SSH struct {
 	// X11 is used to configure X11 forwarding settings
 	X11 *X11 `yaml:"x11,omitempty"`
 
+	// MaybeSSHFileCopying enables or disables remote file operations via SCP/SFTP.
+	// We're using a pointer-to-bool here because the system default is to allow
+	// SCP/SFTP, we need to distinguish between an unset value and a false
+	// value so we can an override unset value with `true`.
+	//
+	// Don't read this value directly: call the SSHFileCopying method
+	// instead.
+	MaybeSSHFileCopying *bool `yaml:"file_copying,omitempty"`
```

Comment:
This is a bit complicated. What do you think about reverting the switch to `disable_file_copying`? Then by default is false (so we don't need a pointer here), when it's set to `true` we will disable the copying.

CC @r0mant

Reply:
`MaybeAllowTCPForwarding` does the same above, and the role options `port_forwarding`, `desktop_clipboard`, `desktop_directory_sharing`, and `create_host_user` all behave the same way, giving some consistency to how this option is implemented both as a role option and as a SSH config option.

I understand your concern, but between the comment that explicitly states not to use the `MaybeSSHFileCopying` directly and the field name starting with `Maybe`, I think it's reasonably easy to use correctly.